onnxruntime/core/framework/instant_op.cc

```diff
@@ -0,0 +1,273 @@
+      }
+      attr->set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_INTS);
+    } else {
+      attr->set_i(ints[0]);
+      attr->set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_INT);
+    }
+  } else if (type == ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING) {
+    if (is_array) {
+      using const_chars = const char*;
+      auto ss = reinterpret_cast<const const_chars*>(data);
+      for (int j = 0; j < len; ++j) {
+        attr->add_strings(std::string{ss[j]});
+      }
+      attr->set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRINGS);
+    } else {
+      auto str = reinterpret_cast<const char*>(data);
+      attr->set_s(std::string{str});
+      attr->set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_STRING);
+    }
+  } else {
+    return Status(common::ONNXRUNTIME, common::FAIL, "Invalid attribute data type.");
+  }
+  *op_attr = reinterpret_cast<OrtOpAttr>(attr.release());
+  return Status::OK();
+}
+
+onnxruntime::Status CreateOperator(const OrtKernelInfo* info,
+                                   const char* op_name,
+                                   const char* domain,
+                                   int version,
+                                   const char** type_constraint_names,
+                                   const ONNXTensorElementDataType* type_constraint_values,
+                                   int type_constraint_count,
+                                   const OrtOpAttr* attr_values,
+                                   int attr_count,
+                                   OrtOp* op) {
+  *op = nullptr;
+  auto kernel_info = reinterpret_cast<const OpKernelInfo*>(info);
+  auto ep = reinterpret_cast<const IExecutionProvider*>(kernel_info->GetExecutionProvider());
+  auto kernel_registry = ep->GetKernelRegistry();
+  const KernelCreateInfo* kernel_create_info{};
+  std::unordered_map<std::string, MLDataType> type_constraint_map;
+  for (int i = 0; i < type_constraint_count; ++i) {
+    ONNX_NAMESPACE::TypeProto proto;
+    proto.mutable_tensor_type()->set_elem_type(type_constraint_values[i]);
+    type_constraint_map[type_constraint_names[i]] = DataTypeImpl::TypeFromProto(proto);
+  }
+  auto status = kernel_registry->TryFindKernel(op_name,
+                                               domain,
+                                               version,
+                                               type_constraint_map,
+                                               ep->Type(),
+                                               &kernel_create_info);
+  ORT_RETURN_IF_ERROR(status);
+  onnxruntime::Node node;
+  for (int i = 0; i < attr_count; ++i) {
+    auto attr_proto = reinterpret_cast<ONNX_NAMESPACE::AttributeProto*>(attr_values[i]);
+    node.AddAttributeProto(*attr_proto);
+  }
+  OpKernelInfo instant_kernel_info(node, KernelDef{}, *ep, {}, {}, {});
```

Comment:
We pass in a default constructed `KernelDef` while constructing an `OpKernelInfo`. Isn't this going to be a problem if/when kernels use `info.GetKernelDef()` to get some info that the implementation may need

Reply:
Good point, calling KernelDefBuilder to build a valid kernel def.